src/midend/mir_emitter.cpp

```diff
@@ -895,6 +943,32 @@ namespace graphit {
         ctx->addElementType(element_type);
     }
 
+
+    std::vector<mir::Var> MIREmitter::emitVars(std::vector<fir::Argument::Ptr> args) {
```

Comment:
The name of this function is a bit confusing. This is specifically emitting vector of Var from Arguments. Maybe give it a more descriptive name?

Reply:
got it